Calculate host fee on pre-tax total

Betree opened the pull request:
Part of https://github.com/opencollective/opencollective/issues/4492

- [x] Add failing test to confirm the issue
- [x] Fix root cause
- [x] Create a migration to update the `HOST_FEE` transactions

Betree commented on test/cron/monthly/host-settlement.test.js:
```diff
@@ -157,40 +203,76 @@ describe('cron/monthly/host-settlement', () => {
+    expect(eurHostSettlementExpense.items[0].description).to.eq('Platform Tips');
+    expect(eurHostSettlementExpense.items[0].amount).to.eq(300e2);
+    expect(eurHostSettlementExpense.items[1].description).to.eq('Shared Revenue');
+    expect(eurHostSettlementExpense.items[1].amount).to.eq(50e2); // 50€ (100€ host fee * 50% host fee share)
```
This one is failing with:

> AssertionError: expected 6050 to equal 5000

Expected: 1000€ contribution × 10% host fee × 50% host fee share = 50€
Actual: (1000€ contribution + 21% VAT) × 10% host fee × 50% host fee share = 60.5€

Which confirms https://github.com/opencollective/opencollective/issues/4492.

Betree commented on test/cron/monthly/host-settlement.test.js:
```diff
@@ -157,40 +205,98 @@ describe('cron/monthly/host-settlement', () => {
+  it('collective balance reflects the taxes & host fees properly', async () => {
+    const balanceAmount = await eurCollective.getBalanceAmount();
+    expect(balanceAmount.currency).to.eq('EUR'); // 1000€ contribution - 121€ (100€ host fee + 21€ VAT)
+    expect(balanceAmount.value).to.eq(900e2); // 1000€ contribution - 100€ host fee
```
This one is also failing:
Expected: 1000€ - 100€ host fee = 900€
Actual: = 1000€ - 121€ host fee = 879€ (because host fee include tax amount)

Betree commented:
Pushed the `scripts/taxes/taxes-on-host-fee.ts` script that will provide the basis for fixing the situation:

```console
opencollective-api:~$ npm run script scripts/taxes/taxes-on-host-fee.ts check-pending
info: Found 1 transactions to update for a total of 1 hosts
info: ==== Host opensource ====
info: 1 transactions to update for a total of $4.84

opencollective-api:~$ DRY_RUN=false npm run script scripts/taxes/taxes-on-host-fee.ts fix 
info: Found 1 transactions to update for a total of 1 hosts
info: ==== Host opensource ====
Updated host fee transactions 52917/52918

opencollective-api:~$ npm run script scripts/taxes/taxes-on-host-fee.ts check-fixed       
info: ==== Host opensource ====
info: Total amount refunded by script: $4.84
info:   - test-event-with-vat-34ed9a9a: $4.84 (Active). Current balance: $76.89
```

znarf reviewed (approved):
Looks good, small lint issue to solve.

Betree commented:
For the record:

```
info: Found 303 transactions to update for a total of 2 hosts
info: ==== Host europe ====
info: Updated host fee transactions 4397879/4397878
info: Updated host fee transactions 4329328/4329327
info: Updated host fee transactions 4321198/4321197
info: Updated host fee transactions 4315153/4315152
info: Updated host fee transactions 4275929/4275928
info: Updated host fee transactions 4271059/4271058
info: Updated host fee transactions 4268980/4268979
info: Updated host fee transactions 4260396/4260395
info: Updated host fee transactions 4245634/4245633
info: Updated host fee transactions 4237958/4237957
info: Updated host fee transactions 4232911/4232910
info: Updated host fee transactions 4232203/4232202
info: Updated host fee transactions 4227978/4227977
info: Updated host fee transactions 4227869/4227868
info: Updated host fee transactions 4226146/4226145
info: Updated host fee transactions 4223131/4223130
info: Updated host fee transactions 4223060/4223059
info: Updated host fee transactions 4223014/4223013
info: Updated host fee transactions 4221918/4221917
info: Updated host fee transactions 4221868/4221867
info: Updated host fee transactions 4221818/4221817
info: Updated host fee transactions 4221777/4221776
info: Updated host fee transactions 4221471/4221470
info: Updated host fee transactions 4221447/4221446
info: Updated host fee transactions 4220037/4220036
info: Updated host fee transactions 4219947/4219946
info: Updated host fee transactions 4209790/4209789
info: Updated host fee transactions 4116160/4116159
info: Updated host fee transactions 4114691/4114689
info: Updated host fee transactions 4111143/4111142
info: Updated host fee transactions 4065459/4065458
info: Updated host fee transactions 4047904/4047903
info: Updated host fee transactions 4045869/4045868
info: Updated host fee transactions 4041161/4041160
info: Updated host fee transactions 4035540/4035539
info: Updated host fee transactions 4035502/4035501
info: Updated host fee transactions 3926039/3926038
info: Updated host fee transactions 3924485/3924484
info: Updated host fee transactions 3920143/3920142
info: Updated host fee transactions 3809507/3809506
info: Updated host fee transactions 3809363/3809362
info: Updated host fee transactions 3794180/3794179
info: Updated host fee transactions 3738692/3738691
info: Updated host fee transactions 3736682/3736681
info: Updated host fee transactions 3732696/3732695
info: Updated host fee transactions 3702196/3702195
info: Updated host fee transactions 3701121/3701120
info: Updated host fee transactions 3680759/3680758
info: Updated host fee transactions 3635132/3635131
info: Updated host fee transactions 3634716/3634715
info: Updated host fee transactions 3634672/3634671
info: Updated host fee transactions 3622155/3622154
info: Updated host fee transactions 3553061/3553060
info: Updated host fee transactions 3543259/3543258
info: Updated host fee transactions 3541485/3541484
info: Updated host fee transactions 3537085/3537084
info: Updated host fee transactions 3501452/3501451
info: Updated host fee transactions 3501390/3501389
info: Updated host fee transactions 3501180/3501179
info: Updated host fee transactions 3501117/3501116
info: Updated host fee transactions 3501039/3501038
info: Updated host fee transactions 3499381/3499380
info: Updated host fee transactions 3496105/3496104
info: Updated host fee transactions 3402918/3402917
info: Updated host fee transactions 3369759/3369758
info: Updated host fee transactions 3344893/3344892
info: Updated host fee transactions 3239192/3239191
info: Updated host fee transactions 3211258/3211257
info: Updated host fee transactions 3209344/3209343
info: Updated host fee transactions 3185391/3185390
info: Updated host fee transactions 3080655/3080654
info: Updated host fee transactions 3074083/3074082
info: Updated host fee transactions 3051077/3051076
info: Updated host fee transactions 3049165/3049164
info: Updated host fee transactions 3026521/3026520
info: Updated host fee transactions 2883359/2883358
info: Updated host fee transactions 2880159/2880158
info: Updated host fee transactions 2855203/2855202
info: Updated host fee transactions 2853157/2853156
info: Updated host fee transactions 2828410/2828409
info: Updated host fee transactions 2816824/2816823
info: Updated host fee transactions 2733635/2733634
info: Updated host fee transactions 2709511/2709510
info: Updated host fee transactions 2582454/2582453
info: Updated host fee transactions 2561473/2561472
info: Updated host fee transactions 2480939/2480938
info: Updated host fee transactions 2438932/2438931
info: Updated host fee transactions 2419292/2419291
info: Updated host fee transactions 2296500/2296499
info: Updated host fee transactions 2275921/2275920
info: Updated host fee transactions 2192574/2192573
info: Updated host fee transactions 2153954/2153953
info: Updated host fee transactions 2132405/2132404
info: Updated host fee transactions 2038819/2038818
info: Updated host fee transactions 2019944/2019943
info: Updated host fee transactions 1997444/1997443
info: Updated host fee transactions 1889229/1889228
info: Updated host fee transactions 1834077/1834076
info: Updated host fee transactions 1713254/1713252
info: Updated host fee transactions 1661921/1661920
info: Updated host fee transactions 1584426/1584425
info: Updated host fee transactions 1531161/1531160
info: Updated host fee transactions 1463882/1463881
info: Updated host fee transactions 1408820/1408819
info: Updated host fee transactions 1347173/1347172
info: Updated host fee transactions 1291937/1291936
info: Updated host fee transactions 1206848/1206847
info: Updated host fee transactions 1171481/1171479
info: Updated host fee transactions 1011329/1011328
info: Updated host fee transactions 973853/973852
info: Updated host fee transactions 925611/925610
info: Updated host fee transactions 938325/938324
info: ==== Host ocnz ====
info: Updated host fee transactions 3689691/3689690
info: Updated host fee transactions 3501606/3501605
info: Updated host fee transactions 3501598/3501597
info: Updated host fee transactions 3501590/3501589
info: Updated host fee transactions 3298304/3298303
info: Updated host fee transactions 2970444/2970443
info: Updated host fee transactions 2970436/2970435
info: Updated host fee transactions 2940723/2940722
info: Updated host fee transactions 2938648/2938647
info: Updated host fee transactions 2938486/2938485
info: Updated host fee transactions 2936301/2936300
info: Updated host fee transactions 2936295/2936294
info: Updated host fee transactions 2936171/2936170
info: Updated host fee transactions 2936163/2936162
info: Updated host fee transactions 2936153/2936152
info: Updated host fee transactions 2936065/2936064
info: Updated host fee transactions 2934280/2934279
info: Updated host fee transactions 2934110/2934109
info: Updated host fee transactions 2931087/2931086
info: Updated host fee transactions 2928512/2928511
info: Updated host fee transactions 2927692/2927691
info: Updated host fee transactions 2910938/2910937
info: Updated host fee transactions 2899464/2899463
info: Updated host fee transactions 2897407/2897406
info: Updated host fee transactions 2875785/2875784
info: Updated host fee transactions 2827284/2827283
info: Updated host fee transactions 2797742/2797741
info: Updated host fee transactions 2797734/2797733
info: Updated host fee transactions 2789036/2789035
info: Updated host fee transactions 2788836/2788835
info: Updated host fee transactions 2787555/2787554
info: Updated host fee transactions 2787196/2787195
info: Updated host fee transactions 2786688/2786687
info: Updated host fee transactions 2780348/2780347
info: Updated host fee transactions 2780162/2780161
info: Updated host fee transactions 2645652/2645651
info: Updated host fee transactions 2645646/2645645
info: Updated host fee transactions 2627022/2627021
info: Updated host fee transactions 2625649/2625648
info: Updated host fee transactions 2625641/2625640
info: Updated host fee transactions 2625633/2625632
info: Updated host fee transactions 2620107/2620106
info: Updated host fee transactions 2498790/2498789
info: Updated host fee transactions 2498611/2498610
info: Updated host fee transactions 2498599/2498598
info: Updated host fee transactions 2484403/2484402
info: Updated host fee transactions 2484395/2484394
info: Updated host fee transactions 2484387/2484386
info: Updated host fee transactions 2484379/2484378
info: Updated host fee transactions 2484371/2484370
info: Updated host fee transactions 2484363/2484362
info: Updated host fee transactions 2468716/2468715
info: Updated host fee transactions 2465878/2465877
info: Updated host fee transactions 2465864/2465863
info: Updated host fee transactions 2465822/2465821
info: Updated host fee transactions 2464648/2464647
info: Updated host fee transactions 2452728/2452727
info: Updated host fee transactions 2210469/2210468
info: Updated host fee transactions 2173266/2173265
info: Updated host fee transactions 2057849/2057848
info: Updated host fee transactions 2056385/2056384
info: Updated host fee transactions 2054017/2054016
info: Updated host fee transactions 2053733/2053732
info: Updated host fee transactions 2053521/2053520
info: Updated host fee transactions 2053363/2053362
info: Updated host fee transactions 2052415/2052414
info: Updated host fee transactions 2052183/2052182
info: Updated host fee transactions 2048848/2048847
info: Updated host fee transactions 2048396/2048395
info: Updated host fee transactions 2046961/2046960
info: Updated host fee transactions 2046953/2046952
info: Updated host fee transactions 2046945/2046944
info: Updated host fee transactions 2046639/2046638
info: Updated host fee transactions 2045461/2045460
info: Updated host fee transactions 2044481/2044480
info: Updated host fee transactions 2044323/2044322
info: Updated host fee transactions 2044257/2044256
info: Updated host fee transactions 2041741/2041740
info: Updated host fee transactions 2041713/2041712
info: Updated host fee transactions 2041669/2041668
info: Updated host fee transactions 2041645/2041644
info: Updated host fee transactions 2041609/2041608
info: Updated host fee transactions 2039461/2039460
info: Updated host fee transactions 2039453/2039452
info: Updated host fee transactions 2039445/2039444
info: Updated host fee transactions 2039437/2039436
info: Updated host fee transactions 1929908/1929907
info: Updated host fee transactions 1929902/1929901
info: Updated host fee transactions 1916819/1916818
info: Updated host fee transactions 1916625/1916624
info: Updated host fee transactions 1913132/1913131
info: Updated host fee transactions 1911738/1911737
info: Updated host fee transactions 1904225/1904224
info: Updated host fee transactions 1904051/1904050
info: Updated host fee transactions 1901719/1901718
info: Updated host fee transactions 1901679/1901678
info: Updated host fee transactions 1900711/1900710
info: Updated host fee transactions 1895687/1895686
info: Updated host fee transactions 1791911/1791910
info: Updated host fee transactions 1779936/1779935
info: Updated host fee transactions 1770863/1770862
info: Updated host fee transactions 1770346/1770345
info: Updated host fee transactions 1766415/1766414
info: Updated host fee transactions 1762976/1762975
info: Updated host fee transactions 1760115/1760114
info: Updated host fee transactions 1756608/1756607
info: Updated host fee transactions 1756400/1756399
info: Updated host fee transactions 1756324/1756323
info: Updated host fee transactions 1756180/1756179
info: Updated host fee transactions 1756164/1756163
info: Updated host fee transactions 1754759/1754758
info: Updated host fee transactions 1751669/1751668
info: Updated host fee transactions 1751661/1751660
info: Updated host fee transactions 1751649/1751648
info: Updated host fee transactions 1735907/1735906
info: Updated host fee transactions 1686803/1686802
info: Updated host fee transactions 1658263/1658262
info: Updated host fee transactions 1655243/1655242
info: Updated host fee transactions 1654687/1654686
info: Updated host fee transactions 1654675/1654674
info: Updated host fee transactions 1654663/1654662
info: Updated host fee transactions 1654655/1654654
info: Updated host fee transactions 1654643/1654642
info: Updated host fee transactions 1654631/1654630
info: Updated host fee transactions 1651701/1651700
info: Updated host fee transactions 1651569/1651568
info: Updated host fee transactions 1651473/1651472
info: Updated host fee transactions 1651215/1651214
info: Updated host fee transactions 1651107/1651106
info: Updated host fee transactions 1650907/1650906
info: Updated host fee transactions 1648667/1648666
info: Updated host fee transactions 1648569/1648568
info: Updated host fee transactions 1648535/1648534
info: Updated host fee transactions 1648513/1648512
info: Updated host fee transactions 1648507/1648506
info: Updated host fee transactions 1648487/1648486
info: Updated host fee transactions 1648429/1648428
info: Updated host fee transactions 1648421/1648420
info: Updated host fee transactions 1648349/1648348
info: Updated host fee transactions 914857/914856
info: Updated host fee transactions 914869/914868
info: Updated host fee transactions 914907/914906
info: Updated host fee transactions 915281/915280
info: Updated host fee transactions 915529/915528
info: Updated host fee transactions 916439/916438
info: Updated host fee transactions 918067/918066
info: Updated host fee transactions 918683/918682
info: Updated host fee transactions 918685/918684
info: Updated host fee transactions 918689/918688
info: Updated host fee transactions 918719/918718
info: Updated host fee transactions 918957/918956
info: Updated host fee transactions 919021/919020
info: Updated host fee transactions 919023/919022
info: Updated host fee transactions 919031/919030
info: Updated host fee transactions 919047/919046
info: Updated host fee transactions 919063/919062
info: Updated host fee transactions 919133/919132
info: Updated host fee transactions 919327/919326
info: Updated host fee transactions 919347/919346
info: Updated host fee transactions 919363/919362
info: Updated host fee transactions 919367/919366
info: Updated host fee transactions 919401/919400
info: Updated host fee transactions 919405/919404
info: Updated host fee transactions 919414/919413
info: Updated host fee transactions 919433/919432
info: Updated host fee transactions 919445/919444
info: Updated host fee transactions 919601/919600
info: Updated host fee transactions 919613/919612
info: Updated host fee transactions 919615/919614
info: Updated host fee transactions 920133/920132
info: Updated host fee transactions 920403/920402
info: Updated host fee transactions 920421/920420
info: Updated host fee transactions 920445/920444
info: Updated host fee transactions 920499/920498
info: Updated host fee transactions 920547/920546
info: Updated host fee transactions 920549/920548
info: Updated host fee transactions 920551/920550
info: Updated host fee transactions 920771/920770
info: Updated host fee transactions 920777/920776
info: Updated host fee transactions 920791/920790
info: Updated host fee transactions 920805/920804
info: Updated host fee transactions 920813/920812
info: Updated host fee transactions 920821/920820
info: Updated host fee transactions 920827/920826
info: Updated host fee transactions 920829/920828
info: Updated host fee transactions 920841/920840
info: Updated host fee transactions 920877/920876
info: Updated host fee transactions 921023/921022
info: Updated host fee transactions 921041/921040
info: Updated host fee transactions 921061/921060
info: Updated host fee transactions 921077/921076
```
